docs/do-not-merge.md

```diff
@@ -0,0 +1,737 @@
+
+## SQL compatible NULL handling
+
+`druid.generic.useDefaultValueForNull` is now disabled by default.
+Druid now differentiates between empty records, such as `' '`, and null records.
+Previously, Druid might treat empty records as empty or null.
+
+This change may impact your query results. For more information, see [SQL compatibility in the upgrade notes](#sql-compatibility).
+
+[#14792](https://github.com/apache/druid/pull/14792)
+
+### JSON and auto column indexer
+
+The `json` type is now equivalent to using `auto` in native JSON-based ingestion dimension specs. Upgrade your ingestion specs to `json` to take advantage of the features and functionality of `auto`, including the following:
+
+- Type specializations including ARRAY typed columns
+- Better support for nested arrays of strings, longs, and doubles
+- Smarter index utilization
+
+`json` type columns created with Druid 28.0.0 are not backwards compatible with Druid versions older than 26.0.0.
+If you upgrade from one of these versions, you can continue to write nested columns in a backwards compatible format (version 4).
+
+For more information, see [Nested column format in the upgrade notes](#nested-column-format).
+
+[#14955](https://github.com/apache/druid/pull/14955)
+[#14456](https://github.com/apache/druid/pull/14456)
+
+## Reset offsets for a supervisor
+
+Added a new API endpoint `/druid/indexer/v1/supervisor/:supervisorId/resetOffsets` to reset specific partition offsets for a supervisor without resetting the entire set.
+This endpoint clears only the specified offsets in Kafka or sequence numbers in Kinesis, prompting the supervisor to resume data reading.
+
+[#14772](https://github.com/apache/druid/pull/14772)
+
+## Removed Hadoop 2
+
+Support for Hadoop 2 has been removed.
+Migrate to SQL-based ingestion or native ingestion if you are using Hadoop 2.x for ingestion today. If migrating to Druid's built-in ingestion is not possible, you must upgrade your Hadoop infrastructure to 3.x+ before upgrading to Druid 28.0.0.
+
+[#14763](https://github.com/apache/druid/pull/14763)
+
+# Additional features and improvements
+
+## SQL-based ingestion
+
+### New ingestion mode for arrays
+
+The MSQ task engine now includes an `arrayIngestMode` that determines how arrays are treated for ingestion. Previously, arrays were ingested as multi-value dimensions (MVDs). They can now be ingested as `ARRAY<STRING>` or numeric arrays instead if the ingest mode is set to `array`. 
+
+The default mode is for MVDs, which is unchanged from previous releases.
+
+[#15093](https://github.com/apache/druid/pull/15093)
+
+### UNNEST functionality
+
+You can now use UNNEST with MSQ queries. This lets you flatten and explode data during batch ingestion. For more information, see the [UNNEST tutorial](https://druid.apache.org/docs/latest/tutorials/tutorial-unnest-arrays/) and [UNNEST documentation](https://druid.apache.org/docs/latest/querying/sql/#unnest).
+
```

Comment:
Should also put a note that the context flag "enableUnnest" is not needed anymore to run queries involving unnest

Reply:
Updated